### PR TITLE
clarify the difference between attributes and  properties

### LINKED ISSUE #3848: Clarify the difference between attribute and property in Container Docu
`<luigi-container viewUrl="/index.html" sandbox-rules='["allow-scripts", "allow-same-origin"]'></luigi-container>`
or
`myContainer.sandboxRules = ["allow-scripts", "allow-same-origin"]`

Are both correct uses. 

**Discuss** of a way to show on the documentation that properties and attributes are usually distinguished with a dash. 

This seems to be a standard [here](https://html.spec.whatwg.org/multipage/custom-elements.html#valid-custom-element-name)

 Discuss where to put the notice, or if it makes even sense to put any notice. 

Whether we put it at the top, of the API page for Container and Compound Container, or for each component separately.

Maybe just a notice 
This document outlines the properties provided by the Luigi Container. <include here a very short statement that mentions the standard for the dash>

### PR DESCRIPTION
**Pull request by smahati**
**Description**

Changes proposed in this pull request:

- documentation suggested for the clarification between attributes and properties

**Related issue(s)**

Fixed #3848

### CODE REVIEW

**Comment by smahati** [comment]
not sure if this is supposed to be in the source code somewhere.

**Review by JohannesDoberer — APPROVED** [review]
LGTM

**Comment by smahati** [comment]
Added snippets for the compound container, if this looks ok I can add snippets to the other API reference as well.

File: docs/luigi-compound-container-api.md [hunk 1]
```diff
@@ -119,6 +191,14 @@ Type: [Object](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Globa
 
 *   **since**: 1.0.0
 
+```js
+myContainer.userSettings = '{foo: bar}'
+```
+
+```HTML
+<luigi-container viewUrl="/index.html" userSettings='{foo: bar}'></luigi-container>
```
Comment (JohannesDoberer) [inline docs/luigi-compound-container-api.md:199]:
```suggestion
<luigi-container viewUrl="/index.html" userSettings='{"language": "de", "theme":"sap_horizon"}'></luigi-container>
```

File: docs/luigi-compound-container-api.md [hunk 2]
```diff
@@ -119,6 +191,14 @@ Type: [Object](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Globa
 
 *   **since**: 1.0.0
 
+```js
+myContainer.userSettings = '{foo: bar}'
```
Comment (JohannesDoberer) [inline docs/luigi-compound-container-api.md:195]:
```suggestion
myContainer.userSettings = {language: 'de', theme: 'sap_horizon'}
```

File: docs/luigi-compound-container-api.md [hunk 3]
```diff
@@ -180,6 +292,14 @@ Type: ([boolean](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Glo
 
 *   **since**: 1.0.0
 
+```js
+myContainer.webcomponent = { webComponentSettings: {type: 'foo', selfRegistered: true, tagName: '#wcMountpoint'}}
```
Comment (JohannesDoberer) [inline docs/luigi-compound-container-api.md:296]:
```suggestion
myContainer.webcomponent = { webComponentSettings: {type: 'module', selfRegistered: true, tagName: 'my-webcomponent'}}
```

File: docs/luigi-compound-container-api.md [hunk 4]
```diff
@@ -180,6 +292,14 @@ Type: ([boolean](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Glo
...
+```
+#TODO: this is probably wrong/not the best way, maybe there is a custom element?
+```HTML
+<luigi-container webcomponent="{ webComponentSettings: {type: 'foo', selfRegistered: true, tagName: '#wcMountpoint'}}"></luigi-container>
```
Comment (JohannesDoberer) [inline docs/luigi-compound-container-api.md:300]:
```suggestion
<luigi-container webcomponent="{ webComponentSettings: {type: 'module', selfRegistered: true, tagName: 'my-webcomponent'}}"></luigi-container>
```

File: docs/luigi-compound-container-api.md [hunk 5]
```diff
@@ -200,6 +328,14 @@ Type: [string](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Globa
 
 *   **since**: NEXT_RELEASE_CONTAINER
 
+```js
+myContainer.theme = 'SAP-Horizon'
```
Comment (JohannesDoberer) [inline docs/luigi-compound-container-api.md:332]:
```suggestion
myContainer.theme = 'sap_horizon'
```

File: docs/luigi-compound-container-api.md [hunk 6]
```diff
@@ -200,6 +328,14 @@ Type: [string](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Globa
...
+```
+
+```HTML
+<luigi-container viewUrl="/index.html" theme='SAP-Horizon'></luigi-container>
```
Comment (JohannesDoberer) [inline docs/luigi-compound-container-api.md:336]:
```suggestion
<luigi-container viewUrl="/index.html" theme='sap_horizon'></luigi-container>
```

File: docs/luigi-compound-container-api.md [hunk 7]
```diff
@@ -38,6 +46,14 @@ Type: [Object](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Globa
 **Meta**
 
 *   **since**: 1.0.0
+TODO: better example
+```js
+myContainer.compoundConfig = {foo: bar}
```
Comment (JohannesDoberer) [inline docs/luigi-compound-container-api.md:51]:
```suggestion
myContainer.compoundConfig = lcc.compoundConfig = {
        renderer: {
          use: 'grid',
          config: {
            columns: '1fr 1fr 1fr 2fr',
            /*rows: '150px',*/
            /*gap: '30px',*/
            layouts: [
              {
                minWidth: 0,
                maxWidth: 600,
                columns: '1fr',
                gap: 0
              },
              {
                minWidth: 600,
                maxWidth: 1024,
                columns: '1fr 1fr',
                gap: '30px'
              }
            ]
          }
        },
        children: [
          {
            viewUrl: 'http://www.examples.com/header.js',
            context: {
              title: 'My Awesome Grid 000',
              description: 'Really awesome'
            },
            layoutConfig: {
              row: '1',
              column: '1 / -1'
            },
            eventListeners: [
              {
                source: 'input1',
                name: 'sendInput',
                action: 'update',
                dataConverter: data => {
                  return 'new text: ' + data;
                }
              }
            ]
          },
          {
            id: 'input1',
            viewUrl: '/helloWorldWC.js',
            context: {
              title: 'Some input',
              instant: true
            }
          },
          {
            viewUrl: 'http://www.examples.com/main.js',
            context: {
              label: 'Another web component'
            }
          },
          {
            viewUrl: '/panelFooter.js',
            context: {
              footer: 'This is the end of awesomeness'
            },
            layoutConfig: {
              column: '1 / -1'
            }
          }
        ]
      };
```

File: docs/luigi-compound-container-api.md [hunk 8]
```diff
@@ -38,6 +46,14 @@ Type: [Object](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Globa
...
+```
+
+```HTML
+<luigi-container viewUrl="/index.html" compoundConfig='{foo: bar}'></luigi-container>
```
Comment (JohannesDoberer) [inline docs/luigi-compound-container-api.md:55]: Not sure if we should provide an example for compould attribute in html

File: docs/luigi-compound-container-api.md [hunk 9]
```diff
@@ -28,6 +28,14 @@ The URL used for the renderer.
 
 Type: [string](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/String)
 
+```js
+myContainer.viewUrl = "/index.html"
+```
+
+```HTML
+<luigi-container viewUrl="/index.html"></luigi-container>
```
Comment (JohannesDoberer) [inline docs/luigi-compound-container-api.md:36]:
```suggestion
<luigi-container viewurl="/index.html"></luigi-container>
```

File: docs/luigi-compound-container-api.md [hunk 10]
```diff
@@ -38,6 +46,83 @@ Type: [Object](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Globa
 **Meta**
 
 *   **since**: 1.0.0
+TODO: better example
+```js
+myContainer.compoundConfig = lcc.compoundConfig = {
```
Comment (JohannesDoberer) [inline docs/luigi-compound-container-api.md:51]:
```suggestion
myContainer.compoundConfig = {
```

File: docs/luigi-compound-container-api.md [hunk 11]
```diff
@@ -38,6 +46,83 @@ Type: [Object](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Globa
...
+            layoutConfig: {
+              column: '1 / -1'
+            }
+          }
+        ]
+      };
+```
+
+```HTML
+<luigi-container viewUrl="/index.html" compoundConfig='{foo: bar}'></luigi-container>
```
Comment (JohannesDoberer) [inline docs/luigi-compound-container-api.md:124]:
```suggestion
<luigi-container viewUrl="/index.html" compound-config='{foo: bar}'></luigi-container>
```

File: docs/luigi-compound-container-api.md [hunk 12]
```diff
@@ -49,6 +134,14 @@ Type: [boolean](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Glob
 
 *   **since**: 1.0.0
 
+```js
+myContainer.deferInit = true
+```
+TODO: is it preferred to assign true or declare it to indicate it's set
+```HTML
+<luigi-container viewUrl="/index.html" deferInit="true"></luigi-container>
```
Comment (JohannesDoberer) [inline docs/luigi-compound-container-api.md:142]:
```suggestion
<luigi-container viewUrl="/index.html" defer-init="true"></luigi-container>
```

File: docs/luigi-compound-container-api.md [hunk 13]
```diff
@@ -59,6 +152,14 @@ Type: [Object](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Globa
 
 *   **since**: 1.0.0
 
+```js
+myContainer.nodeParams = {foo: bar}
+```
+
+```HTML
+<luigi-container viewUrl="/index.html" nodeParams='{"node":"param"}'></luigi-container>
```
Comment (JohannesDoberer) [inline docs/luigi-compound-container-api.md:160]:
```suggestion
<luigi-container viewUrl="/index.html" node-params='{"node":"param"}'></luigi-container>
```

File: docs/luigi-compound-container-api.md [hunk 14]
```diff
@@ -69,6 +170,14 @@ Type: [boolean](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Glob
 
 *   **since**: 1.2.0
 
+```js
+myContainer.noShadow = true
+```
+
+```HTML
+<luigi-container viewUrl="/index.html" noShadow='true'></luigi-container>
```
Comment (JohannesDoberer) [inline docs/luigi-compound-container-api.md:178]:
```suggestion
<luigi-container viewUrl="/index.html" no-shadow='true'></luigi-container>
```

File: docs/luigi-compound-container-api.md [hunk 15]
```diff
@@ -79,6 +188,14 @@ Type: [Object](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Globa
 
 *   **since**: 1.0.0
 
+```js
+myContainer.searchParams = {foo: bar}
+```
+
+```HTML
+<luigi-container viewUrl="/index.html" searchParams='{"search":"param"}'></luigi-container>
```
Comment (JohannesDoberer) [inline docs/luigi-compound-container-api.md:196]:
```suggestion
<luigi-container viewUrl="/index.html" search-params='{"search":"param"}'></luigi-container>
```

File: docs/luigi-compound-container-api.md [hunk 16]
```diff
@@ -89,6 +206,14 @@ Type: [Object](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Globa
 
 *   **since**: 1.0.0
 
+```js
+myContainer.pathParams = {foo: bar}
+```
+
+```HTML
+<luigi-container viewUrl="/index.html" pathParams='{"path":"param"}'></luigi-container>
```
Comment (JohannesDoberer) [inline docs/luigi-compound-container-api.md:214]:
```suggestion
<luigi-container viewUrl="/index.html" path-params='{"path":"param"}'></luigi-container>
```

File: docs/luigi-compound-container-api.md [hunk 17]
```diff
@@ -109,6 +242,14 @@ Type: [Object](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Globa
 
 *   **since**: 1.0.0
 
+```js
+myContainer.clientPermissions = {permission: "adminGroup"}
+```
+
+```HTML
+<luigi-container viewUrl="/index.html" clientPermissions='{"permission": "adminGroup"}'></luigi-container>
```
Comment (JohannesDoberer) [inline docs/luigi-compound-container-api.md:250]:
```suggestion
<luigi-container viewUrl="/index.html" client-permissions='{"permission": "adminGroup"}'></luigi-container>
```

File: docs/luigi-compound-container-api.md [hunk 18]
```diff
@@ -119,6 +260,14 @@ Type: [Object](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Globa
 
 *   **since**: 1.0.0
 
+```js
+myContainer.userSettings = {language: 'de', theme: 'sap_horizon'}
+```
+
+```HTML
+<luigi-container viewUrl="/index.html" userSettings='{"language": "de", "theme":"sap_horizon"}'></luigi-container>
```
Comment (JohannesDoberer) [inline docs/luigi-compound-container-api.md:268]:
```suggestion
<luigi-container viewUrl="/index.html" user-settings='{"language": "de", "theme":"sap_horizon"}'></luigi-container>
```

File: docs/luigi-compound-container-api.md [hunk 19]
```diff
@@ -139,6 +296,14 @@ Type: [string](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Globa
 
 *   **since**: 1.2.0
 
+```js
+myContainer.documentTitle = 'Luigi App'
+```
+
+```HTML
+<luigi-container viewUrl="/index.html" documentTitle='Luigi App'></luigi-container>
```
Comment (JohannesDoberer) [inline docs/luigi-compound-container-api.md:304]:
```suggestion
<luigi-container viewUrl="/index.html" document-title='Luigi App'></luigi-container>
```

File: docs/luigi-compound-container-api.md [hunk 20]
```diff
@@ -150,6 +315,14 @@ Type: [string](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Globa
 
 *   **since**: 1.2.0
 
+```js
+myContainer.hasBack = '/detailsPage'
+```
+
+```HTML
+<luigi-container viewUrl="/index.html" hasBack='/detailsPage'></luigi-container>
```
Comment (JohannesDoberer) [inline docs/luigi-compound-container-api.md:323]:
```suggestion
<luigi-container viewUrl="/index.html" has-back='/detailsPage'></luigi-container>
```

File: docs/luigi-compound-container-api.md [hunk 21]
```diff
@@ -161,6 +334,14 @@ Type: [boolean](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Glob
 
 *   **since**: 1.2.0
 
+```js
+myContainer.dirtyStatus = true
+```
+
+```HTML
+<luigi-container viewUrl="/index.html" dirtyStatus='true'></luigi-container>
```
Comment (JohannesDoberer) [inline docs/luigi-compound-container-api.md:342]:
```suggestion
<luigi-container viewUrl="/index.html" dirty-status='true'></luigi-container>
```

File: docs/luigi-compound-container-api.md [hunk 22]
```diff
@@ -190,6 +379,14 @@ Type: [Array](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global
 
 *   **since**: NEXT_RELEASE_CONTAINER
 
+```js
+myContainer.activeFeatureToggleList = ["enable-foo", "allow-bar"]
+```
+
+```HTML
+<luigi-container viewUrl="/index.html" activeFeatureToggleList='["enable-foo", "allow-bar"]'></luigi-container>
```
Comment (JohannesDoberer) [inline docs/luigi-compound-container-api.md:387]:
```suggestion
<luigi-container viewUrl="/index.html" active-feature-toggle-list='["enable-foo", "allow-bar"]'></luigi-container>
```

File: docs/luigi-compound-container-api.md (hunk 8 above)
Comment (smahati) [reply docs/luigi-compound-container-api.md]: given the size of it, it seems reasonable to not wanting people to do it this way so I removed it

File: container/typings/LuigiContainer.svelte.d.ts [hunk 23]
```diff
@@ -50,78 +66,181 @@ export default class LuigiContainer extends HTMLElement {
...
   /**
    * The path parameters to be passed to the web-component-based micro frontend.
    * @since 1.0.0
+   * 
+   * ```js
+   * myContainer.pathParams = {foo: bar}
+   * ```
+   * 
+   * ```HTML
+   * <luigi-container viewUrl="/index.html" path-params='{"path":"param"}'></luigi-container>
```
Comment (JohannesDoberer) [inline container/typings/LuigiContainer.svelte.d.ts:200]:
```suggestion
   * <luigi-container viewurl="/index.html" path-params='{"path":"param"}'></luigi-container>
```

File: container/typings/LuigiContainer.svelte.d.ts [hunk 24]
```diff
@@ -146,20 +265,44 @@ export default class LuigiContainer extends HTMLElement {
...
    * The hasBack value to be passed to the web-component-based micro frontend.
    * It indicates that there is one or more preserved views. Useful when you need to show a back button.
    * @since 1.2.0
+   * 
+   * ```js
+   * myContainer.hasBack = '/detailsPage'
+   * ```
+   * 
+   * ```HTML
+   * <luigi-container viewUrl="/index.html" has-back='/detailsPage'></luigi-container>
```
Comment (JohannesDoberer) [inline container/typings/LuigiContainer.svelte.d.ts:289]:
```suggestion
   * <luigi-container viewurl="/index.html" has-back></luigi-container>
```

File: container/typings/LuigiContainer.svelte.d.ts [hunk 25]
```diff
@@ -146,20 +265,44 @@ export default class LuigiContainer extends HTMLElement {
...
+   * ```
    */
   hasBack: string;
```
Comment (JohannesDoberer) [inline container/typings/LuigiContainer.svelte.d.ts:292]:
```suggestion
  hasBack: boolean;
```

File: container/typings/LuigiContainer.svelte.d.ts [hunk 26]
```diff
@@ -146,20 +265,44 @@ export default class LuigiContainer extends HTMLElement {
...
    * The dirty status value to be passed to the web-component-based micro frontend.
    * It's used to indicate that there are unsaved changes when navigating away.
    * @since 1.2.0
+   * 
+   * ```js
+   * myContainer.dirtyStatus = true
+   * ```
+   * 
+   * ```HTML
+   * <luigi-container viewUrl="/index.html" dirty-status='true'></luigi-container>
```
Comment (JohannesDoberer) [inline container/typings/LuigiContainer.svelte.d.ts:304]:
```suggestion
   * <luigi-container viewurl="/index.html" dirty-status='true'></luigi-container>
```

**Comment by JohannesDoberer** [comment]
After the changes please run `npm run docu` in /scripts because it generates the docu

**Comment by smahati** [comment]
@JohannesDoberer I added the examples to both files and generated the docs, I can clean it up a bit more but it's ready to be reviewed.

**Review by JohannesDoberer — APPROVED** [review]
LGTM

File: website/docs/.nvmrc [hunk 27]
```diff
@@ -1 +1 @@
-18
```
Comment (JohannesDoberer) [inline website/docs/.nvmrc:1]: Why did you add this and removed the 18?

Comment (smahati) [reply website/docs/.nvmrc:1]: I use nvm and it was causing issues as my v18's minor version was not meeting the minimum requirement. However lts/hydrogen is also v18 but updated to the latest version so I changed it to make it easier but can revert if needed and I can probably update my v18 to the latest version too.

**Review by JohannesDoberer — APPROVED** [review]
